configure.ac

```diff
@@ -404,15 +404,7 @@ AS_IF([test $BUILD_ZLIB = no],
 AS_IF([test x$BUILD_ZLIB = xyes],[
   BUILD_ZLIB=yes
   ZLIB_CPPFLAGS='-I${abs_builddir}/extern/install/zlib/include'
-  AS_CASE([$host_os],
-    [*cygwin*],[
-      ZLIB_LDFLAGS='-L${abs_builddir}/extern/install/zlib/lib/ -lz'
-    ],
-    [*],[
-      ZLIB_LDFLAGS='${abs_builddir}/extern/install/zlib/lib/libz.a'
-    ]
-  )
-
+  ZLIB_LDFLAGS='-L${abs_builddir}/extern/install/zlib/lib/ -Wl,-rpath -Wl,${abs_builddir}/extern/install/zlib/lib/ -lz'
```

Comment:
The one issue with that is that it won't work with `make install` at all. But that is of course a general problem when building with our "builtin zlib". It is less of a problem for our built-in GMP, as that also is using libtool, and so we can  implement a `make install` which properly re-links all libraries and binaries during installation with relatively little effort.

Anyway, I suspect the best long-term fix would be to simply replace the zlib build system with a custom one written by us; I might look into that at some point.

But anyway, if this change fixes a real world issue for somebody/something, let's merge it; in the very worst case, we could still revert it anyway.